internal/io/chunk_writer.go

```diff
@@ -0,0 +1,151 @@
+package io
+
+import (
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+
+	"github.com/goto/optimus-any2any/internal/helper"
+)
+
+type chunkWriter struct {
+	l          *slog.Logger
+	chunkSize  int
+	size       int
+	extension  string
+	skipHeader bool // used only when extension is csv, tsv, or xlsx
+	delimiter  rune // used only when extension is csv, or tsv
+	writer     io.Writer
+
+	// internal use
+	writerTmp io.ReadWriteSeeker // temporary writer to write the data until chunk is full
+	cleanTmp  func()
+	written   int64
+}
+
+var _ io.WriteCloser = (*chunkWriter)(nil)
+var _ WriteFlusher = (*chunkWriter)(nil)
+
+// NewChunkWriter creates a new chunk writer with the specified chunk size and extension
+func NewChunkWriter(l *slog.Logger, writer io.Writer, opts ...Option) *chunkWriter {
+	w := &chunkWriter{
+		l:         l,
```

Comment:
I see logger initiated here but not used, can we add logging at least for debugs only?